backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisSessionManager.java

```diff
@@ -79,7 +79,8 @@ void updateSession(Request request, Response response) {
     }
 
     // check if session state and session user map still in sync
-    if (!session.getSessionUser().equals(session.getDatabase().getActiveUser())) {
+    if (session.getSessionUser() != null
+        && !session.getSessionUser().equals(session.getDatabase().getActiveUser())) {
```

Comment:
Lots of if-else-ing going on here combined with lots of null checking and fall-though cases, could really use unit tests that covers the different cases and describes each case